client/src/components/user-link-metabox/index.js

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from 'react';
+import Autocomplete from 'react-autocomplete';
+import PropTypes from 'prop-types';
+import { Button } from '@wordpress/components';
+
+// Hooks.
+import useDebounce from '../../services/use-debounce';
+
+const UserLinkMetaBox = ({
+  linkUserPlaceholder,
+  linkedToLabel,
+  postId,
+  unlinkLabel,
+  user: rawUser,
+  userAlreadyLinked,
+  usersApiUrl,
+}) => {
+  const [search, setSearch] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+  const [user, setUser] = useState({});
+
+  // Debounce search string from input.
+  const debouncedSearchString = useDebounce(search, 750);
+
+  const doUserSearch = (fragment) => {
+    fetch(
+      `${usersApiUrl}?s=${fragment}&post=${postId}`,
+    )
+      .then((res) => res.json())
+      .then((results) => {
+        setSearchResults(results);
+      });
+  };
+
+  const inputProps = {
+    placeholder: linkUserPlaceholder,
+    onKeyDown: (e) => {
+      // If the user hits 'enter', stop the parent form from submitting.
+      if (13 === e.keyCode) {
+        e.preventDefault();
+      }
+    },
+  };
+
+  useEffect(() => {
+    if ('' !== debouncedSearchString) {
+      doUserSearch(debouncedSearchString);
+    }
+  }, [debouncedSearchString]);
+
+  useEffect(() => {
+    setUser(rawUser);
+  }, []);
+
+  return (
+    <div className="profile-user-link byline-manager-meta-box">
+      <input
+        type="hidden"
+        name="profile_user_link"
+        value={user.id || 0}
+      />
+      {user && user.id && user.name ? (
+        <p className="current-user-link">
+          {`${linkedToLabel} `}
+          <a href={`/wp-admin/user-edit.php?user_id=${user.id}`}>
```

Comment:
I don't think this path will work as expected in multisite installations where sites use subdirectories, a not-uncommon setup. It would link me from `mainsite.com/subsite/wp-admin/...` to `mainsite.com/wp-admin/user-edit.php`, which I might not have permission to access.

I can see that this issue existed in the old component, so I suppose it doesn't need to be a blocker now, but if you're so inclined, it might be fixable by including the URL to the user edit screen in the response from the users API endpoint (unless that endpoint is publicly accessible, in which case, let's not add an admin link to it) or by exporting the base admin URL from the server to the browser.

Reply:
👍 I think this might also be a good improvement. For all intent and purpose, nothing has changed from what we had before other than these components have been renamed and refactored (in some cases).